src/metric/sensors/recorder.cpp

```diff
@@ -85,9 +85,10 @@ Recorder::Recorder(trace::Trace& trace)
         int chip_name_size = sensors_snprintf_chip_name(nullptr, 0, chip);
         assert(chip_name_size > 0);
 
-        std::string chip_name;
-        chip_name.resize(chip_name_size);
-        int res = sensors_snprintf_chip_name(chip_name.data(), chip_name.size(), chip);
```

Comment:
There's no need to use `vector`. The patch simply is:

`int res = sensors_snprintf_chip_name(chip_name.data(), chip_name.size() + 1, chip);`

It looks scary, but is valid since C++11, as `std::string::data()` returns a pointer to a null-terminated array, but `std::string::size()` returns the number of characters without counting the null-terminator. Hence, it is fine to add one to `size()`.